rdtools/normalization.py

```diff
@@ -282,20 +287,74 @@ def irradiance_rescale(irrad, modeled_irrad):
         measured irradiance time series
     modeled_irrad: Pandas Series (numeric)
         modeled irradiance time series
+    max_iterations: (int)
+        The maximum number of times to attempt rescale optimization, default 100.
+        Ignored if method = 'single_opt'
+    method: (str)
+        The caclulation method to use. 'single_opt' implements the irradiance_rescale of
+        rdtools v1.1.3 and earlier. 'iterative' implements a more stable calculation
+        that may yield different results from the single_opt method. Default None issues
+        a warning then uses the iterative calculation.
+
     Returns
     -------
     Pandas Series (numeric): resacaled modeled irradaince time series
     '''
-    def _rmse(fact):
-        rescaled_modeled_irrad = fact * modeled_irrad
-        csi = irrad / rescaled_modeled_irrad
-        filt = (csi >= 0.8) & (csi <= 1.2)
-        rmse = np.sqrt(((rescaled_modeled_irrad[filt] - irrad[filt]) ** 2.0).mean())
-        return rmse
-
-    guess = np.percentile(irrad.dropna(), 90) / np.percentile(modeled_irrad.dropna(), 90)
-    min_result = minimize(_rmse, guess, method='Nelder-Mead')
-    factor = min_result['x'][0]
-
-    out_irrad = factor * modeled_irrad
-    return out_irrad
+
+    if method is None:
+        warnings.warn("The underlying calculations for irradiance_rescale have changed "
+                      "which may affect results. To revert to the version of irradiance_rescale "
+                      "from rdtools v1.1.3 or earlier, use method = 'single_opt'. ")
```

Comment:
This is a nice way to keep the code backwards compatible while warning that the default behavior has changed. In this situation, I think either UserWarning or SyntaxWarning are appropriate.

FutureWarning is most applicable in the case where a code feature is slated to be removed in a future version -- if we decide that we no longer wish to support `method='single-opt'`, for example.